File: SaladimHelper/ModFolder/Loenn/entities/explodeFocusArea.lua
local area = {}

area.name = "SaladimHelper/ExplodeFocusArea"
area.depth = 6000
area.fillColor = { 0.7294, 0.8745, 0.9568, 0.2 }
area.borderColor = { 1.0, 1.0, 1.0 }
area.placements = {
    name = "normal",
    data = {
        width = 8,
        height = 8,
        focusType = 1,
        rotation = 0
    }
}

area.fieldInformation = {
    focusType = {
        options = {
            ["EightWay"] = 0,
            ["FourWay"] = 1,
            ["DiagonalFourWay"] = 2,
            ["HorizontalTwoWay"] = 3,
            ["VerticalTwoWay"] = 4,
        },
        editable = false,
        fieldType = "integer"
    }
}

return area

File: SaladimHelper/Entities/ExplodeFocusArea.cs
using Celeste.Mod.Entities;

namespace Celeste.Mod.SaladimHelper.Entities;

[CustomEntity("SaladimHelper/ExplodeFocusArea"), Tracked]
public partial class ExplodeFocusArea : Entity
{
    public enum FocusType
    {
        EightWay,
        FourWay,
        DiagonalFourWay,
        HorizontalTwoWay,
        VerticalTwoWay
    }

    private readonly FocusType focusType;
    private readonly float rotation;

    public ExplodeFocusArea(EntityData data, Vector2 offset)
        : this(data.Position + offset, data.Width, data.Height, (FocusType)data.Int("focusType"), MathHelper.ToRadians(data.Float("rotation")))
    {
    }

    public ExplodeFocusArea(Vector2 position, int width, int height, FocusType focusType, float rotation)
        : base(position)
    {
        this.focusType = focusType;
        this.rotation = rotation;
        Collider = new Hitbox(width, height);
    }
}
